docs/monitoring.md

```diff
@@ -0,0 +1,294 @@
+# Metrics {#ovms_docs_metrics}
+
+## Introduction
+
+This document describes how to use metrics endpoint in the OpenVINO Model Server. They can be applied for:
+
+- Providing performance and utilization statistics for monitoring and benchmarking purposes
+
+- Auto scaling of the model server instances in Kubernetes and OpenShift based on application related metrics
+
+> **NOTE**: currently, Metrics feature is released as a preview feature.
+
+Built-in metrics allows tracking the performance without any extra logic on the client side or using network traffic monitoring tools like load balancers or reverse-proxies.
+
+It also exposes metrics which are not related to the traffic. 
+
+For example statistics of the inference execution queue, model runtime parameters etc. They can also track the usage based on model version, API type or requested endpoint methods.
+
+OpenVINO Model Server metrics are compatible with Prometheus standard [Prometheus metrics detailed description](https://prometheus.io/docs)
+
+They are exposed on the `/metrics` endpoint.
+
+## Available metrics families
+
+Metrics from default list are enabled with the `metrics_enabled` flag or json configuration.
+
+However, you can enable also additional metrics by listing all the metrics you want to enable in the `metric_list` flag or json configuration.
+
+
+DEFAULT
+
+   ```bash
+         gauge       ovms_streams                        Number of OpenVINO execution streams.
+         gauge       ovms_current_requests               Number of inference requests currently in process.
+         counter     ovms_requests_success               Number of successful requests to a model or a DAG.
+         counter     ovms_requests_fail                  Number of failed requests to a model or a DAG.
+         histogram   ovms_request_time_us                Processing time of requests to a model or a DAG.
+         histogram   ovms_inference_time_us              Inference execution time in the OpenVINO backend.
+         histogram   ovms_wait_for_infer_req_time_us     Request waiting time in the scheduling queue.
+   ```
+
+ADDITIONAL
+
+   ```bash
+         gauge   ovms_infer_req_queue_size       Inference request queue size (nireq).
+         gauge   ovms_infer_req_active           Number of currently consumed inference request from the processing queue.
+   ```
+
+## List of available metrics labels
```

Comment:
why don't we list the labels in the same table like above in additional column. It could be just "api, interface,method,name,version"
We could include also a table with description of the labels:
label_name,  description, values

Reply:
Done.